lib/rubocop/cop/lint/format_parameter_mismatch.rb

```diff
@@ -0,0 +1,92 @@
+# encoding: utf-8
+
+module RuboCop
+  module Cop
+    module Lint
+      # This lint sees if there is a mismatch between the number of
+      # expected fields for format/sprintf/#% and what is actually
+      # passed as arguments.
+      #
+      # @example
+      #
+      #   format('A value: %s and another: %i', a_value)
+      #
+      class FormatParameterMismatch < Cop
+        MSG = 'Number of fields (%i) mismatches number of arguments (%i).'
```

Comment:
Maybe you should mention the name of the method in the message (`format` or `sprintf`). It will make it clearer I think.